portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,19 +31,22 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. TODO: modify this file to handle 
+ * comments data */
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
+    int maxComments = Integer.parseInt(request.getParameter("max-comments"));
```

Comment:
You don't have to do this in this change but something to keep in mind.

This is good place to have a try catch block (see example 2 in this page - https://alvinalexander.com/java/edu/qanda/pjqa00010.shtml). For example if I try to go to https://kyraa-step-2020.appspot.com/comments?max-comments=dsada I get a Server Error (because Integer.parseInt will not work for "dsada"). The way to handle this with useful error message would be to change your return json value to also include a error status and message and in your javascript you will check the error status and display comments if no error and display the error message if there is an error.